### PR TITLE
refactor(dev-infra): refactor commit-message files

### PR DESCRIPTION
**Pull request by josephperrott**
Refactor the commit-message files to be consistent with how other ng-dev tooling
is structured.

### CODE REVIEW

File: dev-infra/commit-message/config.ts [hunk 1]
```diff
@@ -8,6 +8,15 @@
 
 import {assertNoErrors, getConfig, NgDevConfig} from '../utils/config';
 
+
+/**
+ * The source triggering the git commit message creation.
+ * As described in: https://git-scm.com/docs/githooks#_prepare_commit_msg
+ */
+export type CommitMsgSource = 'message'|'template'|'merge'|'squash'|'commit';
```
Comment (devversion) [inline dev-infra/commit-message/config.ts:16]: I realize you put this here as it's the most common file of `commit-message`, though it seems not 100% related to config. Maybe just move into a separate file?

File: dev-infra/commit-message/restore-commit-message/cli.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  fileEnvVariable: string[];
+}
+
+/** Builds the command. */
+function builder(yargs: Argv) {
+  return yargs.option('file-env-variable' as 'fileEnvVariable', {
+    type: 'string',
+    array: true,
+    conflicts: ['file'],
+    required: true,
```
Comment (devversion) [inline dev-infra/commit-message/restore-commit-message/cli.ts:25]: Should we use `demandOption` instead? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/yargs/index.d.ts#L675-L679

Looks like `required` is deprecated

File: dev-infra/commit-message/restore-commit-message/cli.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+async function handler({fileEnvVariable}: Arguments<RestoreCommitMessageOptions>) {
+  restoreCommitMessage(fileEnvVariable[0], fileEnvVariable[1] as CommitMsgSource);
+}
+
+/** yargs command module describing the command.  */
+export const RestoreCommitMessageModule: CommandModule<{}, RestoreCommitMessageOptions> = {
+  handler,
+  builder,
+  command: 'restore-commit-message-draft',
+  describe: false,
```
Comment (devversion) [inline dev-infra/commit-message/restore-commit-message/cli.ts:49]: Should we just add descriptions while we are at it?

File: dev-infra/commit-message/restore-commit-message/cli.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+export interface RestoreCommitMessageOptions {
+  fileEnvVariable: string[];
+}
+
+/** Builds the command. */
+function builder(yargs: Argv) {
+  return yargs.option('file-env-variable' as 'fileEnvVariable', {
+    type: 'string',
+    array: true,
+    conflicts: ['file'],
```
Comment (devversion) [inline dev-infra/commit-message/restore-commit-message/cli.ts:24]: I know this hasn't changed as part of this PR, but what does this resolve to? I don't see a `file` option.

File: dev-infra/commit-message/restore-commit-message/cli.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import {restoreCommitMessage} from './restore-commit-message';
+
+export interface RestoreCommitMessageOptions {
+  fileEnvVariable: string[];
+}
+
+/** Builds the command. */
+function builder(yargs: Argv) {
+  return yargs.option('file-env-variable' as 'fileEnvVariable', {
```
Comment (devversion) [inline dev-infra/commit-message/restore-commit-message/cli.ts:21]: For a follow-up: It looks like this option is not only the env variable (as in the other command) right? Should there be a better name for it or two options for the env variable and source?

File: dev-infra/commit-message/validate-range/cli.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    description: 'The range of commits to check, e.g. --range abc123..xyz456',
+    demandOption: '  A range must be provided, e.g. --range abc123..xyz456',
+    type: 'string',
+    requiresArg: true,
+  });
+}
+
+/** Handles the command. */
+async function handler({range}: Arguments<ValidateRangeOptions>) {
+  // If on CI, and not pull request number is provided, assume the branch
```
Comment (devversion) [inline dev-infra/commit-message/validate-range/cli.ts:32]: typo not part of this PR but just mentioning it if you come across it anyway w/ other feedback: `not` -> `no`

File: dev-infra/commit-message/wizard/cli.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+async function handler(args: Arguments<WizardOptions>) {
+  await runWizard(args);
+}
+
+/** yargs command module describing the command.  */
+export const WizardModule: CommandModule<{}, WizardOptions> = {
+  handler,
+  builder,
+  command: 'wizard <filePath> [source] [commitSha]',
+  describe: false,
```
Comment (devversion) [inline dev-infra/commit-message/wizard/cli.ts:51]: Can we add a description here too?

**Review by devversion — APPROVED** [review]
Nice cleanup!! Just a few comments

File: dev-infra/commit-message/restore-commit-message/cli.ts (hunk 3 above)
Comment (josephperrott) [reply dev-infra/commit-message/restore-commit-message/cli.ts:49]: Having `describe` as false removes it from the help messaging.  This is intentional.

File: dev-infra/commit-message/wizard/cli.ts (hunk 7 above)
Comment (josephperrott) [reply dev-infra/commit-message/wizard/cli.ts:51]: Having `describe` as false removes it from the help messaging.  This is intentional.

File: dev-infra/commit-message/restore-commit-message/cli.ts (hunk 4 above)
Comment (josephperrott) [reply dev-infra/commit-message/restore-commit-message/cli.ts:24]: Its a bad copy-paste from when it was originally created.  Removed.

File: dev-infra/commit-message/restore-commit-message/cli.ts (hunk 5 above)
Comment (josephperrott) [reply dev-infra/commit-message/restore-commit-message/cli.ts:21]: Will create followup

File: dev-infra/commit-message/restore-commit-message/cli.ts (hunk 3 above)
Comment (devversion) [reply dev-infra/commit-message/restore-commit-message/cli.ts:48]: Should we add a comment? I can guess it's because it's just used by the Husky hooks.

Comment (josephperrott) [reply dev-infra/commit-message/restore-commit-message/cli.ts:48]: Added as comment.
